frontend/src/components/widgets/Multiselect/Multiselect.tsx

```diff
@@ -155,14 +167,34 @@ class Multiselect extends React.PureComponent<Props, State> {
   }
 
   private onChange = (params: OnChangeParams): void => {
-    const newState = this.generateNewState(params)
-    this.setState(newState, () => this.commitWidgetValue({ fromUi: true }))
+    if (this.maxSelections) {
```

Comment:
@AnOctopus , I can simplify this logic from this:
`if (this.maxSelections) {
      if (params.type === "clear") {
        const newState = this.generateNewState(params)
        this.setState(newState, () => this.commitWidgetValue({ fromUi: true }))
      } else if (params.type === "remove") {
        const newState = this.generateNewState(params)
        this.setState(newState, () => this.commitWidgetValue({ fromUi: true }))
      } else if (params.type === "select") {
        if (this.state.value.length < this.maxSelections) {
          const newState = this.generateNewState(params)
          this.setState(newState, () =>
            this.commitWidgetValue({ fromUi: true })
          )
        }
      }
    } else {
      const newState = this.generateNewState(params)
      this.setState(newState, () => this.commitWidgetValue({ fromUi: true }))
    }`
to this:

`// Do NOT change state when max selection is set, we are selecting, and value.length >= this.maxSelections
    if(!this.maxSelections || params.type !== "select" || this.state.value.length < this.maxSelections) {
      const newState = this.generateNewState(params)
      this.setState(newState, () => this.commitWidgetValue({ fromUi: true }))
    }`
    
    Which one do you think should go into the code base?

Reply:
I think the above is easier to follow but the next one is cleaner and less lines of code thought I did leave a comment that I think would be helpful so I think the second one is better.